src/view.c

```diff
@@ -19,6 +19,24 @@
 #define LAB_FALLBACK_WIDTH  640
 #define LAB_FALLBACK_HEIGHT 480
 
+bool
+view_matches_query(struct view *view, struct view_query *query)
```

Comment:
Very neat. This is nice AND logic for multiple queries.

We ought to consider combining this with `view_matches_criteria()` in `window-rules.c`, but can do that later. The way you've written it is better. https://github.com/labwc/labwc/blob/206db43786e066f7de3755a11263a971b4ed77be/src/window-rules.c#L46

Two considerations:

- Use `strcasecmp()`
- Use `match_glob()`